eos/statistics/log-likelihood.cc

```diff
@@ -1073,19 +1082,39 @@ namespace eos
 
             virtual std::string as_string() const
             {
-                std::string result = "UniformBound: current value = " + stringify(cache[id]);
+                std::string result = "UniformBound: ";
+                result += "bound = " + stringify(_bound) + " +- " + stringify(_uncertainty);
 
                 return result;
             }
 
             virtual double evaluate() const
             {
-                return cache[id];
+                double saturation = 0.0;
+
+                for (auto i = 0u ; i < _number_of_observables ; ++i)
+                {
+                    saturation += _cache[_ids[i]];
+                }
+
+                if (saturation < 0.0)
+                {
+                    throw InternalError("Contribution to the uniform bound must be positive; found to be negative!");
+                }
+                else if ((0.0 <= saturation) && (saturation < _bound))
+                {
+                    return 0.0;
+                }
+                else
+                {
+                    // add a gaussian like penalty
+                    return - 0.5 * power_of<2>((saturation - _bound) / _uncertainty);
+                }
             }
 
             virtual unsigned number_of_observations() const
             {
-                return 0.0;
+                return _number_of_observables;
```

Comment:
Hm, is this correct? The number of observations is relevant to a chi^2 test statistic, only, correct? Hence the ``0.0`` previously, even though there was always at least one observable bounded.

Reply:
It was incorrect, thanks!

Reply:
This interface is deprecated anyway, since we now use the ``TestStatistic`` type.

Reply:
Ok, but I set it back to zero.

Reply:
Thanks!